### DISCUSSION TITLE
Single Tenant Database Approach

### DISCUSSION THREAD

**Discussion by keithmclaughlin** [spatie/laravel-multitenancy#332] [General]
If you want to use a single database for tenants, do you still need a landlord database, or can you just have a tenants table in the tenants database?

If you did use a single database (which included the tenants table - for landlord use), how would you go about differentiating between a landlord login and a tenant login?

But if you created a separate landlord database, I presume you need to add a users table for logins.

What if, as a landlord, you wanted to be able to do more than just add/edit tenants, like view statistics about tenant use, how would you go about making the connection to the tenant database?

Am I over complicating the whole multi-tenancy design or are these genuine concerns?

Any advice is greatly appreciated :)

**Comment by masterix21** [comment]
> If you want to use a single database for tenants, do you still need a landlord database, or can you just have a tenants table in the tenants database?

You can use one database, with a `tenants` table.

> If you did use a single database (which included the tenants table - for landlord use), how would you go about differentiating between a landlord login and a tenant login?

There is not only one solution: it depends on your approach (domain isolation or a field "Your tenant" in the login?)

> But if you created a separate landlord database, I presume you need to add a users table for logins.

Use a `tenant_id` field in your tenant-aware tables.

> What if, as a landlord, you wanted to be able to do more than just add/edit tenants, like view statistics about tenant use, how would you go about making the connection to the tenant database?

Take a look at https://github.com/spatie/laravel-stats

> Am I over complicating the whole multi-tenancy design or are these genuine concerns?

I don't know what's your project and your idea, but I'm sure that simplicity wins, ever: promote it.
